### PR TITLE
Ff133 dialog toggle beforetoggle events

### PR DESCRIPTION
**Pull request by hamishwillee**
FF133 adds support for `<dialog>` elements to respond to `HTMLElement` events `toggle` and `beforetoggle`. 

This updates the docs to make it clear:
- `toggle` fired on details element, dialog element, popovers
- `beforetoggle` fired on dialog element, popovers (details to follow)

`HTMLDetailsElement` toggle event page removed as that event is actually `HTMLElement` `toggle` event fired on the details element.

Added/updated some examples to make it more clear

One problem is that the events are defined in HTMLElement onto other elements, so they aren't shown as part of those elements in the normal boilerplate - i.e. the list of events. I have mentioned the events in examples.
I have also added BCD entries for the target elements on the events, and will include those as compat data once/if those get accepted.

Related docs work #36536

### CODE REVIEW

File: files/en-us/web/api/htmldetailselement/index.md [hunk 1]
```diff
@@ -26,10 +26,79 @@ _No specific method; inherits methods from its parent, {{domxref("HTMLElement")}
...
 
-Listen to this event using [`addEventListener()`](/en-US/docs/Web/API/EventTarget/addEventListener) or by assigning an event listener to the `oneventname` property of this interface.
-
-- [`toggle`](/en-US/docs/Web/API/HTMLDetailsElement/toggle_event)
-  - : Fired when the `open`/`closed` state of a {{HtmlElement("details")}} element is toggled.
+_Inherits events from its parent interface, {{DOMxRef("HTMLElement")}}._
+
+## Examples
+
+### Log chapters as they are opened and closed
```
Comment (hamishwillee) [inline files/en-us/web/api/htmldetailselement/index.md:33]: Example stolen from the removed HTMLDetailsElement toggle event page.

File: files/en-us/web/api/htmlelement/beforetoggle_event/index.md [hunk 2]
```diff
@@ -32,23 +32,27 @@ A {{domxref("ToggleEvent")}}. Inherits from {{domxref("Event")}}.
...
+
 ```js
 const popover = document.getElementById("mypopover");
 
 // ...
 
 popover.addEventListener("beforetoggle", (event) => {
   if (event.newState === "open") {
-    console.log("Popover is being shown");
+    console.log("Popover is about to be shown");
```
Comment (hamishwillee) [inline files/en-us/web/api/htmlelement/beforetoggle_event/index.md:45]: More work to be done here. A `beforetoggle` can be prevented by calling preventDefault(). Need and example and info about that. Still testing.

File: files/jsondata/GroupData.json [hunk 3]
```diff
@@ -811,10 +811,11 @@
         "Element: focus",
         "Element: select",
         "Element: show",
-        "HTMLDetailsElement: toggle",
         "HTMLDialogElement: cancel",
         "HTMLDialogElement: close",
         "HTMLElement: input",
+        "HTMLElement: beforetoggle",
+        "HTMLElement: toggle",
```
Comment (hamishwillee) [inline files/jsondata/GroupData.json:818]: FYI these are HTML dom elements. They are ALSO in the Popover API entry, so you see them in that sidebar, which is nice. Pity we can't also make them visible on the dialog!

File: files/en-us/web/api/htmldetailselement/index.md [hunk 4]
```diff
@@ -26,10 +26,79 @@ _No specific method; inherits methods from its parent, {{domxref("HTMLElement")}
...
+    <summary>Chapter II</summary>
+    Philosophy in Fortune's name replies to Boethius' reproaches, and proves
+    that the gifts of Fortune are hers to give and to take away.
+  </details>
+  <details id="ch3">
+    <summary>Chapter III</summary>
+    Boethius falls back upon his present sense of misery. Philosophy reminds him
+    of the brilliancy of his former fortunes.
+  </details>
+</section>
```
Comment (pepelsbey) [inline files/en-us/web/api/htmldetailselement/index.md:63]:
```suggestion
<section id="summaries">
  <p>Chapter summaries:</p>
  <details id="ch1">
    <summary>Chapter I</summary>
    Philosophy reproves Boethius for the foolishness of his complaints against
    Fortune. Her very nature is caprice.
  </details>
  <details id="ch2">
    <summary>Chapter II</summary>
    Philosophy in Fortune's name replies to Boethius' reproaches, and proves
    that the gifts of Fortune are hers to give and to take away.
  </details>
  <details id="ch3">
    <summary>Chapter III</summary>
    Boethius falls back upon his present sense of misery. Philosophy reminds him
    of the brilliancy of his former fortunes.
  </details>
</section>
<aside id="log">
  <p>Open chapters:</p>
  <div data-id="ch1" hidden>I</div>
  <div data-id="ch2" hidden>II</div>
  <div data-id="ch3" hidden>III</div>
</aside>
```

File: files/en-us/web/api/htmldetailselement/index.md [hunk 5]
```diff
@@ -26,10 +26,79 @@ _No specific method; inherits methods from its parent, {{domxref("HTMLElement")}
...
+```
+
+#### CSS
+
+```css
+body {
+  display: flex;
+  flex-direction: row-reverse;
```
Comment (pepelsbey) [inline files/en-us/web/api/htmldetailselement/index.md:71]:
```suggestion
```

File: files/en-us/web/api/htmlelement/toggle_event/index.md [hunk 6]
```diff
@@ -33,13 +32,50 @@ A {{domxref("ToggleEvent")}}. Inherits from {{domxref("Event")}}.
...
+const logElement = document.querySelector("#log");
+function log(text) {
+  logElement.innerText = `${logElement.innerText}${text}\n`;
+  logElement.scrollTop = logElement.scrollHeight;
+}
+```
+
+#### JavaScript
+
+The code gets adds an event listener for the `toggle` event and logs the state.
```
Comment (pepelsbey) [inline files/en-us/web/api/htmlelement/toggle_event/index.md:74]:
```suggestion
The code adds an event listener for the `toggle` event and logs the state.
```

**Review by pepelsbey — COMMENTED** [review]
Looks good! Thank you :)

I'd suggest removing reversed layout from the demo and reversing the HTML instead. Reversed layout is always a potential accessibility issue. In this case, it also complicates the code for no reason.

**Comment by hamishwillee** [comment]
Thanks @pepelsbey - I accepted all your suggestions. Note, I didn't consider the example much at all, but you're right it is better this way.

**Review by pepelsbey — APPROVED** [review]
Nice!

**Pull request merged**